### PR TITLE
DOC: Blogpost_KD_Week7

### PR DESCRIPTION
**Pull request by deka27**
Summary:
This pull request introduces a blog post for the 8th week of GSOC. It aims to provide insights and reflections for future work as well.

Key Changes:

Added a new blog post file in the posts/2024 directory.

### CODE REVIEW

File: posts/2024/2024_07_15_kaustav_week_7.rst [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+Difficulties & Challenges
+-------------------------
+
+While fixing the PRs, these were some of my thoughts
+
+.. _#3257:
+
+1. `PR #3257 <https://github.com/dipy/dipy/pull/3257>`_
+- Most of the issues were fixed thanks to the review by my mentor. One clarification still remains about the **fit_method**. I was receiving error as mentioned below, to fix it it I applied an `else` clause for `fit_method` in `FreeWaterTensorModel`.
```
Comment (itellaetxe) [inline posts/2024/2024_07_15_kaustav_week_7.rst:32]: Add missing article (**an**): `...I was receiving error...` -> `...I was receiving an error...`

Remove duplicated **it**:   `...below, to fix it it...` -> `...below, to fix it...`

Add missing article (**the**): `...'else' clause for 'fit_method'...` -> `...'else' clause for the 'fit_method'...`

File: posts/2024/2024_07_15_kaustav_week_7.rst [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+                self.design_matrix, data, S0, *self.args, **self.kwargs
+            )
+    E       TypeError: 'str' object is not callable
+
+    dipy/reconst/fwdti.py:161: TypeError
+
+.. _#3254:
+
+2. `PR #3254 <https://github.com/dipy/dipy/pull/3254>`_
+- I am still having issue with this PR, where applying decorator to two specific functions gives me errors. I am assuming the problem arrives because my decorator is not receiving correct signature from the cython function. I was able to resolve it locally through `cython.binding` method but it didn't work out for the CI.
```
Comment (itellaetxe) [inline posts/2024/2024_07_15_kaustav_week_7.rst:92]: Need plural form: `...having issue with...` -> `...having issues with...`

Missing article (**the**): `...where applying decorator...` -> `...here applyng the decorator...`

Missing article (**a** or **the**): `...receiving correct signature...` -> `...receiving a/the signature...`

Missing article (**the**): `...locally through 'cython.binding' method...` -> `...locally through the 'cython.binding' method...`

File: posts/2024/2024_07_15_kaustav_week_7.rst [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+
+.. _#3288:
+
+3. `PR #3288 <https://github.com/dipy/dipy/pull/3288>`_
+- Very good progress, I have a few little doubts to complete this PR. I will try to ask them in the next meeting session with my mentor. Also for some reason the external libraries are not getting loaded even with correct implementation of `lazy_loader <https://pypi.org/project/lazy-loader/>`__ package. Hopefully I can resolve this soon.
```
Comment (itellaetxe) [inline posts/2024/2024_07_15_kaustav_week_7.rst:97]: Style (**ask** -> **treat**): `I will try to ask them in the next meeting session...` -> `I will try to treat them in the next meeting session...`

Missing article**s** (**the**): `...even with correct implementation of 'lazy_loader'...` -> `...even with the correct implementation of the 'lazy_loader'...`

File: posts/2024/2024_07_15_kaustav_week_7.rst [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+
+Next Week
+---------
+
+Next week I am planning to take care of some documentation issues that are raised in the repository. Also I have some tasks assigned by my mentor towards improvements of DIPY.
```
Comment (itellaetxe) [inline posts/2024/2024_07_15_kaustav_week_7.rst:102]: (Optional) Could you **briefly** specify which tasks? Just for some context :)

**Review by itellaetxe — COMMENTED** [review]
Good work Kaustav! It is good to see that you progress in your project.

I added some comments about the style. Minor stuff:

File: posts/2024/2024_07_15_kaustav_week_7.rst (hunk 1 above)
Comment (deka27) [reply posts/2024/2024_07_15_kaustav_week_7.rst:32]: Thanks

File: posts/2024/2024_07_15_kaustav_week_7.rst (hunk 2 above)
Comment (deka27) [reply posts/2024/2024_07_15_kaustav_week_7.rst:92]: Thanks

**Review by robinroy03 — APPROVED** [review]

**Review by skoudoro — APPROVED** [review]
All good, merging

**Pull request merged**